src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -160,9 +160,12 @@ public void actionPerformed(ActionEvent e) {
     userAddButton.addActionListener(new ActionListener() {
       @Override
       public void actionPerformed(ActionEvent e) {
-        final String s = (String) JOptionPane.showInputDialog(
-            UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
-            null, null, "");
+        String s = "";
+        try {
+          s = OAuth2.getEmail();
+        } catch (Exception exception) {
+          System.out.println(exception);
```

Comment:
I'm sure you thought about this, so just to confirm, it'd be nice to show the error message in the GUI.